File: Easy/Python/387. First Unique Character in a String.py
class Solution(object):
    def firstUniqChar(self, s):
        myDict = {}

        for c in s:
            if c not in myDict:
                myDict[c] = 1
            else:
                myDict[c] = myDict.get(c) + 1
        
        for i in range(len(s)):
            if myDict[s[i]] == 1:
                return i
        
        return -1

File: Easy/JavaScript/387. First Unique Character in a String.js
var firstUniqChar = function(s) {
    let map = new Map();

    for(let c of s){
        if(!map.has(c))
            map.set(c, 1);
        else
            map.set(c, map.get(c) + 1);
    }

    for(let i = 0; i < s.length; i++){
        if(map.get(s.charAt(i)) == 1)
            return i;
    }

    return -1;
};
